quodlibet/ext/songsmenu/playlist.py

```diff
@@ -118,6 +118,9 @@ def __get_files(self, songs, dir_path, relative=False):
                 path = song('~filename')
                 if relative:
                     path = relpath(path, dir_path)
+                    if path.startswith("#"):
+                        # avoid lines starting with '#' which don't work with M3U
+                        path = os.path.join(os.curdir, path)
```

Comment:
Is `os.curdir` always relative though?

Reply:
`os.path.join(".", path)` might work though

A test would be good here...

Reply:
It's [documented as](https://docs.python.org/3/library/os.html?highlight=os%20curdir#os.curdir) "The constant string used by the operating system to refer to the current directory. This is '.' for Windows and POSIX." That seems pretty clear-cut to me. Specifically, "constant string" suggests it can't be implemented any other way, e.g. using the process working directory.

> A test would be good here...

Yeah ok, I can do that.

Reply:
ok cool

Reply:
Ok, I added some tests.